### PR TITLE
[Improve][Connector-V2] update vectorType

### LINKED ISSUE #7445: [Improve][API] Optimize vectorType
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Necessary to adjust intermediate state of float-vector to byteBuffer and unify vectorType.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by corgy-w**
### Purpose of this pull request
issue : #7445 

need wait pr: #7375 to be completed, some minor emendations are required.

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.common.utils;
+
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+
+public class BufferUtil {
```
Comment (Hisoka-X) [inline seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java]:
```suggestion
public class ByteBufferUtil {
```

Comment (corgy-w) [reply seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java]: I suggest we call it BufferUtil. We can add other NIO buffer tools here too.

Comment (Hisoka-X) [reply seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java]: ok

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    public static ByteBuffer shortArrayToByteBuffer(Short[] shortArray) {
```
Comment (Hisoka-X) [inline seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java]:
```suggestion
    public static ByteBuffer toByteBuffer(Short[] shortArray) {
```

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        }
+
+        // Compatible compilation and running versions are not consistent
+        // Flip the buffer to prepare for reading
+        ((Buffer) byteBuffer).flip();
+
+        return byteBuffer;
+    }
+
+    public static Short[] byteBufferToShortArray(ByteBuffer byteBuffer) {
```
Comment (Hisoka-X) [inline seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java]:
```suggestion
    public static Short[] toShortArray(ByteBuffer byteBuffer) {
```

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        Short[] shortArray = new Short[byteBuffer.capacity() / 2];
+
+        for (int i = 0; i < shortArray.length; i++) {
+            shortArray[i] = byteBuffer.getShort();
+        }
+
+        return shortArray;
+    }
+
+    public static ByteBuffer floatArrayToByteBuffer(Float[] floatArray) {
```
Comment (Hisoka-X) [inline seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java]:
```suggestion
    public static ByteBuffer toByteBuffer(Float[] floatArray) {
```

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        for (float value : floatArray) {
+            byteBuffer.putFloat(value);
+        }
+
+        ((Buffer) byteBuffer).flip();
+
+        return byteBuffer;
+    }
+
+    public static Float[] byteBufferToFloatArray(ByteBuffer byteBuffer) {
```
Comment (Hisoka-X) [inline seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java]:
```suggestion
    public static Float[] toFloatArray(ByteBuffer byteBuffer) {
```

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+        Float[] floatArray = new Float[byteBuffer.capacity() / 4];
+
+        for (int i = 0; i < floatArray.length; i++) {
+            floatArray[i] = byteBuffer.getFloat();
+        }
+
+        return floatArray;
+    }
+
+    public static ByteBuffer doubleArrayToByteBuffer(Double[] doubleArray) {
```
Comment (Hisoka-X) [inline seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java]:
```suggestion
    public static ByteBuffer toByteBuffer(Double[] doubleArray) {
```

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+        for (double value : doubleArray) {
+            byteBuffer.putDouble(value);
+        }
+
+        ((Buffer) byteBuffer).flip();
+
+        return byteBuffer;
+    }
+
+    public static Double[] byteBufferToDoubleArray(ByteBuffer byteBuffer) {
```
Comment (Hisoka-X) [inline seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java]:
```suggestion
    public static Double[] toDoubleArray(ByteBuffer byteBuffer) {
```

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+        Double[] doubleArray = new Double[byteBuffer.capacity() / 8];
+
+        for (int i = 0; i < doubleArray.length; i++) {
+            doubleArray[i] = byteBuffer.getDouble();
+        }
+
+        return doubleArray;
+    }
+
+    public static ByteBuffer intArrayToByteBuffer(Integer[] intArray) {
```
Comment (Hisoka-X) [inline seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java]:
```suggestion
    public static ByteBuffer toByteBuffer(Integer[] intArray) {
```

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+        for (int value : intArray) {
+            byteBuffer.putInt(value);
+        }
+
+        ((Buffer) byteBuffer).flip();
+
+        return byteBuffer;
+    }
+
+    public static Integer[] byteBufferToIntArray(ByteBuffer byteBuffer) {
```
Comment (Hisoka-X) [inline seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java]:
```suggestion
    public static Integer[] toIntArray(ByteBuffer byteBuffer) {
```

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java (hunk 1 above)
Comment (Hisoka-X) [inline seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java]: Please add UT for `BufferUtil`.
PS: change `BufferUtil` to `BufferUtils`

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/oceanbase/OceanBaseMysqlJdbcRowConverter.java [hunk 10]
```diff
@@ -89,11 +89,13 @@ public SeaTunnelRow toInternal(ResultSet rs, TableSchema tableSchema) throws SQL
                     fields[fieldIndex] = JdbcFieldTypeUtils.getFloat(rs, resultSetIndex);
                     break;
                 case FLOAT_VECTOR:
-                    List<Float> vector = new ArrayList<>();
-                    for (Object o : (Object[]) rs.getObject(fieldIndex)) {
-                        vector.add(Float.parseFloat(o.toString()));
+                    Object[] objects = (Object[]) rs.getObject(fieldIndex);
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/oceanbase/OceanBaseMysqlJdbcRowConverter.java:92]: cc @xxsc0529 as well.

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java (hunk 1 above)
Comment (corgy-w) [reply seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/BufferUtil.java]: get

**Review by Hisoka-X — APPROVED** [review]
LGTM. Thanks @corgy-w

**Review by hailin0 — APPROVED** [review]

**Pull request merged**